### PR TITLE
Feat gdv 195 maplayers new config

### PR DESCRIPTION
**Pull request by CindyvdVries**

### CODE REVIEW

File: layouts/default.vue [hunk 1]
```diff
@@ -91,28 +92,43 @@ export default {
...
+      const rasterLayer = getRasterLayer()
+      rasterLayer.source.tiles = this.activeRasterData
+      return rasterLayer
     },
     vectorLayers() {
       const vectorLayers = this.activeVectorData
       const defaultVectorLayer = getVectorLayer()
-      vectorLayers.forEach(layer => {
+
+      const merged = _(vectorLayers)
```
Comment (SiggyF) [inline layouts/default.vue]: comments? source?

File: plugins/bootstrap.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import isEmpty from 'lodash/fp/isEmpty'
 function loadThemes(store) {
   const themes = store.state.map.themes
   if (isEmpty(themes)) {
-    store.dispatch('map/loadThemes')
+    store.dispatch('map/loadDatasets')
```
Comment (SiggyF) [inline plugins/bootstrap.js]: why?

File: store/map/datasets.js [hunk 3]
```diff
@@ -11,31 +11,41 @@ const emptyLocationsObject = () => ({})
...
-const getSpatialData = pipe([get('spatial'), getOrEmptySpatial])
+const getRasterData = pipe([get('raster'), getOrEmptyRaster])
 const getMetadata = pipe([get('metadata'), getOrEmptyMetadata])
 
 export const state = () => ({})
 
 export const mutations = {
   addDatasetVector(state, data) {
-    const id = data.id
+    const id = get('id', data)
```
Comment (SiggyF) [inline store/map/datasets.js]: `_(data).get('id')` or `_.get(data, 'id')`

File: store/map/datasets.js [hunk 4]
```diff
@@ -11,31 +11,41 @@ const emptyLocationsObject = () => ({})
...
+    if (!id) return
     if (!state[id]) Vue.set(state, id, {})
     const vectorData = getVectorData(state[id])
     Vue.set(
       state[id],
       'vector',
-      merge(vectorData, { mapboxLayer: data.mapboxLayer }),
+      // TODO: make generic by looping over vectorLayer
+      merge(vectorData, { mapboxLayer: get('vectorLayer.mapboxLayer', data) }),
```
Comment (SiggyF) [inline store/map/datasets.js]: create variable

File: store/map/datasets.js [hunk 5]
```diff
@@ -11,31 +11,41 @@ const emptyLocationsObject = () => ({})
...
     )
   },
-  addDatasetSpatial(state, data) {
-    const id = data.id
+  addDatasetRaster(state, data) {
+    const id = get('id', data)
+    // If id already has a rasterLayer or url is not correct return
+    if (!id || data.rasterLayer.url === null) return
```
Comment (SiggyF) [inline store/map/datasets.js]: When would this occur?  Separate configuration check?

File: store/map/index.js [hunk 6]
```diff
@@ -66,58 +65,37 @@ export const mutations = {
...
+      })
 
-    const processTheme = applyTo([
-      storeTheme,
-      storeMetadata,
-      storeSpatial,
-      storeVector,
-    ])
+      val.themes.forEach(theme => commit('themes/addTheme', theme))
+      val.datasets.forEach(set => {
```
Comment (SiggyF) [inline store/map/index.js]: comment

File: store/map/index.js [hunk 7]
```diff
@@ -210,13 +188,9 @@ export const getters = {
...
 
-  activeTimestamp(state, { activeSpatialData }) {
-    if (
-      activeSpatialData &&
-      activeSpatialData.length &&
-      activeSpatialData[0] !== ''
-    ) {
-      const str = activeSpatialData[0]
+  activeTimestamp(state, { activeRasterData }) {
+    if (get('[0]', activeRasterData)) {
```
Comment (SiggyF) [inline store/map/index.js]: `_.head(activeRasterData)` `_.first(activeRasterData)`, `_.get(activeRasterData, 0)`?

File: test/unit/store/map/index/actions.spec.js [hunk 8]
```diff
@@ -161,20 +156,10 @@ describe('loadPointDataForLocation', () => {
...
-      locationCode: 'ef',
-      startTime: moment()
-        .subtract(3, 'days')
-        .format('YYYY-MM-DDTHH:mm:ssZ'),
-    })
     expect(commit.mock.calls[0]).toEqual([
       'datasets/addDatasetPointData',
       {
-        id: 'cd',
+        id: 'ab',
```
Comment (SiggyF) [inline test/unit/store/map/index/actions.spec.js]: I prefer `abc`!

File: layouts/default.vue (hunk 1 above)
Comment (CindyvdVries) [reply layouts/default.vue]: comments in https://github.com/openearth/dgds-front-end/pull/30

File: store/map/datasets.js (hunk 5 above)
Comment (CindyvdVries) [reply store/map/datasets.js]: check inside index.js (where datasets are read out)

File: store/map/index.js (hunk 6 above)
Comment (CindyvdVries) [reply store/map/index.js]: check

**Pull request merged**